File: Servicios/Seguridad.vb
Imports System.Collections.Generic
Imports System.Text
Imports System.Security.Cryptography
Imports System.IO
Imports Excepciones
Public Class Seguridad

    Public Shared Function Encriptar(ByVal Dato As String) As String
        Dim TextoEncriptado As Byte()
        Dim DatoEntrada As Byte() = Encoding.UTF8.GetBytes(Dato)
        Using AlgAes As New RijndaelManaged
            AlgAes.Key = Encoding.ASCII.GetBytes("{M[3]}{R(C)}{4(D)}{3[R]}{50(63)}")
            AlgAes.IV = Encoding.ASCII.GetBytes("{.[N](3){T}[VS]}")

            Using MS As New MemoryStream(DatoEntrada.Length)
                Using CS As New CryptoStream(MS, AlgAes.CreateEncryptor(AlgAes.Key, AlgAes.IV), CryptoStreamMode.Write)
                    CS.Write(DatoEntrada, 0, DatoEntrada.Length)
                End Using

                TextoEncriptado = MS.ToArray()
            End Using
        End Using
        Return Convert.ToBase64String(TextoEncriptado)
    End Function


    Public Shared Function Desencriptar(ByVal Dato As String) As String
        Dim TextoDesencriptado As String = String.Empty
            Dim DatoEntrada As Byte() = Convert.FromBase64String(Dato)
            Using AlgAes As New RijndaelManaged
                AlgAes.Key = Encoding.ASCII.GetBytes("{M[3]}{R(C)}{4(D)}{3[R]}{50(63)}")
                AlgAes.IV = Encoding.ASCII.GetBytes("{.[N](3){T}[VS]}")

                Dim MS As New MemoryStream(DatoEntrada)
                Dim CS As New CryptoStream(MS, AlgAes.CreateDecryptor(AlgAes.Key, AlgAes.IV), CryptoStreamMode.Read)
                Dim SR As New StreamReader(CS)

                TextoDesencriptado = SR.ReadToEnd()
            End Using
            Return TextoDesencriptado
    End Function


    Public Shared Function EncriptarMD5(ByVal Dato As String) As String
        Dim md5 As MD5 = New MD5CryptoServiceProvider()
        Dim resultado As Byte()
        resultado = md5.ComputeHash(Encoding.ASCII.GetBytes(Dato))

        Dim Cadena As New StringBuilder()
        For i As Integer = 0 To resultado.Length - 1
            Cadena.Append(resultado(i).ToString("x2"))
        Next
        Return Cadena.ToString()

    End Function


End Class

File: Servicios/Seguridad.cs
using System;
using System.IO;
using System.Security.Cryptography;
using System.Text;

namespace Servicios
{
    public class Seguridad
    {
        public static string Encriptar(string Dato)
        {
            byte[] TextoEncriptado;
            var DatoEntrada = Encoding.UTF8.GetBytes(Dato);
            using (var AlgAes = new RijndaelManaged())
            {
                AlgAes.Key = Encoding.ASCII.GetBytes("{M[3]}{R(C)}{4(D)}{3[R]}{50(63)}");
                AlgAes.IV = Encoding.ASCII.GetBytes("{.[N](3){T}[VS]}");
                using (var MS = new MemoryStream(DatoEntrada.Length))
                {
                    using (var CS = new CryptoStream(MS, AlgAes.CreateEncryptor(AlgAes.Key, AlgAes.IV), CryptoStreamMode.Write))
                    {
                        CS.Write(DatoEntrada, 0, DatoEntrada.Length);
                    }

                    TextoEncriptado = MS.ToArray();
                }
            }

            return Convert.ToBase64String(TextoEncriptado);
        }

        public static string Desencriptar(string Dato)
        {
            string TextoDesencriptado = string.Empty;
            var DatoEntrada = Convert.FromBase64String(Dato);
            using (var AlgAes = new RijndaelManaged())
            {
                AlgAes.Key = Encoding.ASCII.GetBytes("{M[3]}{R(C)}{4(D)}{3[R]}{50(63)}");
                AlgAes.IV = Encoding.ASCII.GetBytes("{.[N](3){T}[VS]}");
                var MS = new MemoryStream(DatoEntrada);
                var CS = new CryptoStream(MS, AlgAes.CreateDecryptor(AlgAes.Key, AlgAes.IV), CryptoStreamMode.Read);
                var SR = new StreamReader(CS);
                TextoDesencriptado = SR.ReadToEnd();
            }

            return TextoDesencriptado;
        }

        public static string EncriptarMD5(string Dato)
        {
            MD5 md5 = new MD5CryptoServiceProvider();
            byte[] resultado;
            resultado = md5.ComputeHash(Encoding.ASCII.GetBytes(Dato));
            var Cadena = new StringBuilder();
            for (int i = 0, loopTo = resultado.Length - 1; i <= loopTo; i++)
                Cadena.Append(resultado[i].ToString("x2"));
            return Cadena.ToString();
        }
    }
}